datasets/flwr_datasets/resplitter/divide_resplitter.py

```diff
@@ -0,0 +1,286 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""DivideResplitter class for Flower Datasets."""
+import collections
+import warnings
+from typing import Dict, List, Optional, Union, cast
+
+import datasets
+from datasets import DatasetDict
+
+
+class DivideResplitter:
+    """Dive existing split(s) of the dataset and assign them custom names.
+
+    Create new `DatasetDict` with new split names corresponding the percentages of data
+    and custom names.
+
+    Parameters
+    ----------
+    divide_config: Union[Dict[str, int], Dict[str, float], Dict[str, Dict[str,
+    int]], Dict[str, Dict[str, float]]]
+        If single level dictionary with keys - the new split names and the values of int
+        = number of samples, float - fraction of the split. The fractions do not have
+        to sum up to 1.0. The order of matter = the first key will get fraction_1
+        starting from the beginning of the dataset.
```

Comment:
```suggestion
        If single level dictionary, keys represent the split names. If values are: int, they
        represent the number of samples in each split; float, they represent the fraction
        of the total samples assigned to that split. These fractions do not have
        to sum up to 1.0. The order of values (either int or float) matter: the first key 
        will get the first split starting from the beginning of the dataset, and so on.
```

Reply:
I tried to make it a bit clearer. But let me know if this is the exact meaning you had in mind.